lib/vendor/lime/lime.php

```diff
@@ -191,6 +191,29 @@ public function ok($exp, $message = '')
   }
 
   /**
+   * Compares two values and returns true if they are equal
+   *
+   * @param mixed  $exp1    left value
+   * @param mixed  $exp2    right value
+   * @return bool
+   */
+  private function equals($exp1, $exp2)
+  {
+    if (is_object($exp1) || is_object($exp2)) {
+      $value = $exp1 === $exp2;
```

Comment:
Why not `return $exp1 === $exp2;` ? Same for others condition, you can directly return the value

Reply:
Because *lime* relies on `==` comparison. If we do `===` it will require to *fix* a lot more test cases.

Reply:
Hum, you didn't get what I told.

Why not return early instead of using `$value`?

```php
if (is_object($exp1) || is_object($exp2)) {
  return $exp1 === $exp2;
} else if (is_float($exp1) && is_float($exp2)) {
  return abs($exp1 - $exp2) < self::EPSILON;
} else if (is_string($exp1) && is_numeric($exp1) || is_string($exp2) && is_numeric($exp2)) {
  return $exp1 == $exp2;
} else if (is_string($exp1) || is_string($exp2)) {
  return (string) $exp1 === (string) $exp2;
}

return $exp1 == $exp2;
```

Reply:
> a lot more test cases.

I'll check the impact and get back with numbers.

Reply:
j0k3r, done that way you can also get rid of the else's.

Reply:
Yup

Reply:
Ah, I see. You're right. I think I just tried to do least possible changeset preserving the original `$value` approach. 

Let me change that.

Reply:
Done. Thanks for the heads up.